src/modules/emote_menu/components/Emotes.jsx

```diff
@@ -0,0 +1,121 @@
+import Button from 'rsuite/lib/Button/index.js';
+import VirtualizedList from './VirtualizedList.jsx';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/emotes.module.css';
+import Emote from './Emote.jsx';
+
+const ROW_HEIGHT = 36;
+const WINDOW_HEIGHT = 308;
+const TOTAL_COLS = 7;
+
+function Emotes({onClick, onHover, section, onSection}) {
+  const wrapperRef = useRef(null);
+  const [, setUpdated] = useState(false);
+
+  const renderRow = useCallback(
+    ({key, style, index}) => {
+      const row = emoteStore.getRow(index);
+      return emoteStore.isHeader(index) ? (
+        <div key={key} style={style} className={styles.header}>
+          <Icon>{row.icon}</Icon>
+          {row.displayName}
+        </div>
+      ) : (
+        <div key={key} style={style} className={styles.row}>
+          {row.map((emote) => (
+            <Emote key={emote.code} emote={emote} onClick={() => onClick(emote)} onMouseOver={() => onHover(emote)} />
+          ))}
+        </div>
+      );
+    },
+    [onHover, onClick]
+  );
+
+  const handleHeaderChange = useCallback((index) => {
+    const header = emoteStore.getRow(index);
+    if (header != null) {
+      onSection(header.id);
+    }
+  });
+
+  useEffect(() => {
+    function callback() {
+      setUpdated((prev) => !prev);
+    }
+    emoteStore.on('updated', callback);
+
+    return () => {
+      emoteStore.off('updated', callback);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!section.scrollTo) return;
+    const index = emoteStore.getHeaderIndexById(section.eventKey);
+    if (index != null) {
+      wrapperRef.current.scrollTo(0, index * ROW_HEIGHT);
+    }
+  }, [section]);
+
+  return (
+    <VirtualizedList
+      stickyRows={emoteStore.headers}
+      rowHeight={ROW_HEIGHT}
+      windowHeight={WINDOW_HEIGHT}
+      totalRows={emoteStore.totalRows}
+      renderRow={renderRow}
+      className={styles.emotesContainer}
+      onHeaderChange={handleHeaderChange}
+      ref={wrapperRef}
+    />
+  );
+}
+
+function SearchedEmotes({search, onHover, onClick}) {
+  const emotes = useMemo(() => emoteStore.search(search), [search]);
+
+  const renderRow = useCallback(
+    ({key, style, index}) => {
+      const row = emotes.slice(index * TOTAL_COLS, (index + 1) * TOTAL_COLS);
+      return (
+        <div key={key} style={style} className={styles.row}>
+          {row.map(({item}) => (
+            <Emote key={item.code} emote={item} onClick={() => onClick(item)} onMouseOver={() => onHover(item)} />
+          ))}
+        </div>
+      );
+    },
+    [onHover, onClick]
+  );
+
+  if (emotes.length === 0) {
+    return (
+      <div className={styles.empty}>
+        <p>We could not find any results...</p>
+        <Button
+          appearance="primary"
+          target="_blank"
+          rel="noreferrer"
+          href={`https://betterttv.com/emotes/shared/search?query=${search}`}>
+          Search Shared Emotes
```

Comment:
is this an actionable cta? what would a user be able to do from there? if it's so they can add to their own channel, maybe only surface it when in that context.. or Pro user.. etc.

Reply:
Probably wasn't a good idea anyway, if a user is searching for an emote they probably know what they're looking for.. not exploring emotes.